mapreduce new

plutoshe opened the pull request.

plutoshe committed:
mapreduce: add mapreduce taskbuilder

plutoshe committed:
mapreduce: initialize mapreduce

Introduce old code,
Add worker task.

plutoshe committed:
mapreduce: user mapper grpc server

user mapper grpc server definition
define the message for user mapper function,
allow user Emit their result
Also, supplement a sample server/client code

plutoshe committed:
mapreduce: user reducer grpc server definition finished

Define the message for user reducer function,
allow user Collect their result.
Also, supplement a sample server/client code

plutoshe committed:
mapreduce: update proto structure, the master, worker task

plutoshe committed:
mapreduce : change the layout MapreduceWorkerStatusPath

Change to MapreduceNodeStatusPath

plutoshe committed:
mapreduce: update a temporary version of master-worker task

plutoshe committed:
mapreduce: implement master logic of grab work mechanism

plutoshe committed:
mapreduce: roughly finished mapper phase

Mapper milestone.

plutoshe committed:
mapreduce: update grab work mechanism method

Worker watch worker status to decide whether grab new work or not

plutoshe committed:
mapreduce: roughly finshed reducer phase

Reducer milestone.

plutoshe committed:
mapreduce: add work mechanism etcd layout

plutoshe committed:
mapreduce: supplement the code before mapper phase

plutoshe committed:
mapreduce: update the work config, the method to transmit work config

plutoshe committed:
mapreduce: roughly check the logic of mapper/reducer procedure

plutoshe committed:
mapreduce: supplement initialization of master task state

And change the mapper/reducer server(client) file name

plutoshe committed:
mapreduce: fix logic bug in initialization

plutoshe committed:
mapreduce: refactor the IO of mapper/reducer procedure

xiang90 commented:
Is this ready for review?

hongchaodeng commented:
I would like to see the design doc to be agreed upon and finished completely first:

https://docs.google.com/document/d/15pT8tzvGbuxlPEKB9901D26da2_wEn-zr3OpylQqDE4/edit?usp=sharing

I think the lack of consensus on arch design impedes development so far, especially when there are a lot of new change.

plutoshe commented:
Not yet, I haven't finished the main code.

plutoshe committed:
mapreduce: fix some syntax bugs

plutoshe committed:
mapreduce: move grpc user server/client to a dir, create example code

plutoshe committed:
mapreduce: pass the build phase of ex.go

xiaoyunwu commented:
@Hongchao,

Mapreduce is an application mostly, its interface to user is fairly simple.
So it is ok for us to first implement something that works, and then worry
about how to improve the implementation.

That said, @Pluto, please update design doc so that it is easy for other to
get a high level understanding of what we try to achieve.

Xiaoyun

On Tue, Jun 2, 2015 at 1:45 PM, PlutoShe notifications@github.com wrote:

> Not yet, I haven't finished the main code.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/taskgraph/mr/pull/1#issuecomment-107814630.

plutoshe committed:
mapreduce: add user program simulator code

plutoshe committed:
mapreduce: fix workId consistency bug

plutoshe committed:
mapreduce: pass through the mapper procedure phase

plutoshe committed:
mapreduce: examine mapper application successful

Finish test of mapper application.

plutoshe committed:
mapreduce: examine reducer application successfully

plutoshe committed:
mapreduce: add example script

plutoshe committed:
mapreduce: change NodeNum attr of Work config to WorkerNum

plutoshe committed:
mapreduce: resolve conflict of work mechanism between tasks

xiang90 commented:
Let me know when this is ready for a review... the pr is getting bigger...

plutoshe commented:
Sure, when it finish, I will notify in mail

plutoshe committed:
mapreduce: aggregate two different type server into one

In sample code, we use a flag variable to identify which sample is
running.

plutoshe committed:
mapreduce: move interface to interface directory, change package deps

plutoshe committed:
mapreduce: add docker initialization config

plutoshe committed:
mapreduce: solve non-local package deps

plutoshe committed:
mapreduce: change user program to docker

plutoshe committed:
mapreduce: test process sentence

a test for process a sentence whose length is between 3 and 30

plutoshe committed:
mapreduce: update the utf8 formation in process sentence server

plutoshe committed:
mapreduce: finished the whole test of process sentence

plutoshe committed:
mapreduce: update process setence(big data) shell program

plutoshe committed:
mapreduce: add background attr to user commandline

Modify the taskgraph deps to plutoshe repo

plutoshe committed:
mapreduce: add grpc/general benchmark

plutoshe committed:
mapreduce: add stream grpc benchmark

plutoshe committed:
mapreduce: change mr mapper to streaming grpc

plutoshe committed:
mapreduce: change the proto to streaming reducer proto

plutoshe committed:
mapreduce: divide worker_task program, update reducer part test

divide worker_task into user_program, process_procedure, and
woker_task_message_process part.
add process_sentence_reducer test part, update server/client program

plutoshe committed:
mapreduce: update proto of kvpair

update key/value attr to message format

plutoshe committed:
mapreduce: framework follows the change of proto

modify the communication rules in framework, to adjust the change of
proto

plutoshe committed:
mapreduce: update docker file

plutoshe committed:
mapreduce: update user program docker java config

plutoshe committed:
mapreduce: update the buffer size

plutoshe committed:
mapreduce: fix Server Broken EOF error

start a goroutine to stop old docker grpc user server, new connection
is being closing

plutoshe committed:
mapreduce: fix "transport is closing" bug

It always failed because we did not wait the docker run image
successed before.

plutoshe committed:
mapreduce: add user mapper example

plutoshe committed:
mapreduce: achieve reducer phase example

plutoshe committed:
mapreduce: add example program config README

plutoshe committed:
mapreduce: update the goroutine mode to process mode

plutoshe committed:
mapreduce: fix dir remove bug

Fix etcd dir remove bug.
Before this commit, this programmer will remove etcd dir in
controller/task process.
Besides, this commit redirect the stdout/stderr of process to a
logfile directory

plutoshe committed:
mapreduce: add logdir config

When run example code in config, add log directory property.

plutoshe committed:
mapreduce: remove sample code

plutoshe committed:
mapreduce: modify deps

Remove deps on taskgraph/taskgraph, substitute with plutoshe/taskgraph

plutoshe committed:
mapreduce: change etcd property

From ETCDBIN to ETCDURL, indicate the etcd exposed ip address,
instead of Old ETCD which represents the executive program path.

plutoshe committed:
mapreduce: add config generator

Add go script to generator involved configuration for application.

plutoshe committed:
mapreduce: fix method error

Due to adapt java grpc change, I changed proto package name from
"proto" to "mapreduce". Which causes rpc error 12(could not find
"proto.Master/GetWork")

This commit modifies the proto.Master to mapreduce.Master.

plutoshe committed:
mapreduce: reduceTaskConfig update

Add log dir property to reducerTaskConfig

plutoshe committed:
mapreduce: update "phase" flag

Update "phase" flag, improve performance

plutoshe committed:
mapreduce: omit clean etcd func err check

plutoshe committed:
mapreduce: add "sudo" to docker exec command
